src/Common/Factory/AbstractFactoryRule.php

```diff
@@ -47,4 +47,18 @@ protected function getClassName(AbstractNode $node): string
 
         return $node->getFullQualifiedName();
     }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $node
+     *
+     * @return bool
+     */
+    protected function isMethodDeprecated(MethodNode $node): bool
```

Comment:
We might have to make this a trait and also use it in other rule :)
But for now this is fine.